CMakeLists.txt

```diff
@@ -1,17 +1,22 @@
 cmake_minimum_required(VERSION 2.8)
 project(mcu)
 
+
+#-----------------------------------------------------------------------------
+# Create version header file
+
 find_package(Git)
 if(GIT_FOUND)
-  execute_process(COMMAND git "rev-parse" "--short" "HEAD" OUTPUT_VARIABLE GIT_COMMIT_HASH WORKING_DIRECTORY ${CMAKE_SOURCE_DIR} OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND git "describe" "--tags" OUTPUT_VARIABLE GIT_COMMIT_HASH WORKING_DIRECTORY ${CMAKE_SOURCE_DIR} OUTPUT_STRIP_TRAILING_WHITESPACE)
 else()
   set(GIT_COMMIT_HASH "Git not found.")
 endif()
+set(VERSION_STRING "${GIT_COMMIT_HASH}")
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/version.c.in
```

Comment:
nit: there is no need for a `version.c`, could be directly a `version.h` and `version.h.in`.
The `version.h.in` could be `const char *DIGITAL_BITBOX_VERSION = "@VERSION_STRING@";`